ftw/zipextract/zipextracter.py

```diff
@@ -0,0 +1,289 @@
+from AccessControl import Unauthorized
+import os
+from plone import api
+from Products.CMFCore.interfaces import IFolderish
+from plone.i18n.normalizer.interfaces import IIDNormalizer
+from tempfile import NamedTemporaryFile
+from ftw.zipextract.interfaces import IFolderCreator
+from ftw.zipextract.interfaces import IFileCreator
+from ftw.zipextract.interfaces import IFile
+from ftw.zipextract import _
+from ftw.zipextract import logger
+import zipfile
+from ZODB.POSException import ConflictError
+from zope import component
+try:
+    from zope.app.container.interfaces import INameChooser
+except ImportError:
+    from zope.container.interfaces import INameChooser
+
+
+class _FileNode():
+
+    def __init__(self, parent_folder, filename, fileid, info):
+        self.name = _(filename)
```

Comment:
Why are you trying to translate the filename?

Reply:
Hum, I think I wanted to escape the html for safety here and thought the `MessageFactory` also did that. Should I use something like `escape` from `xml.sax.saxutils`, because folder and file names are used in the template? Or is that  unnecessary and plone takes care of it?

Reply:
The page tempalte engine takes care of that:
given `foo="a<b"`
safe: `<div tal:content="foo" />` => `<div>a&lt;b</div>`
not safe: `<div tal:content="structure foo" />` => `<div>a<b</div>`

So you want to use `structure` whenever you are 100% sure that the inserted value is valid and safe HTML. Normally you just do not add `structure` and you are safe when using `tal`-commands (content, replace, attributes, etc).

Reply:
Right, I'd seen that at some point... Thanks!